gradle/suppress.gradle

```diff
@@ -5,59 +5,78 @@ import org.w3c.dom.Document
-        addNewSuppressions(unSuppressedCves, suppressionsDoc)
-        suppressionsFile.write(XmlUtil.serialize(cleanXml(suppressionsDoc)))
+        // Suppression file must be specified as 'suppressionFile=...' in build.gradle.
+        File suppressionFile = project.file(dependencyCheck.getSuppressionFile())
+        String suppressionXml = new String(suppressionFile.readBytes(), "UTF-8")
+        Element suppressionDoc = DOMBuilder.parse(new StringReader(suppressionXml)).getDocumentElement()
+        addNewSuppression(unSuppressedCves, suppressionDoc)
+        suppressionFile.write(XmlUtil.serialize(cleanXml(suppressionDoc)))
     }
 }
 
-static void addNewSuppressions(unSuppressedCves, suppressions) {
-    unSuppressedCves.each {
-        String cveName = it
-        Document document = suppressions.getOwnerDocument()
-        Element suppressTag = document.createElement("suppress")
-        suppressTag.appendChild(document.createElement("notes")).appendChild(document.createTextNode("Ticket TBC"))
-        suppressTag.appendChild(document.createElement("cve")).appendChild(document.createTextNode(cveName))
-        suppressions.appendChild(suppressTag)
-    }
+static void addNewSuppression(unSuppressedCves, suppressionDoc) {
+    // Find a specified phrase in notes element and amend text content with new cve
+    XPathExpression xpathExp = XPathFactory.newInstance().newXPath().compile(
+            "//*[local-name()='notes']")
+    NodeList noteNodes =  xpathExp.evaluate(suppressionDoc, XPathConstants.NODESET) as NodeList
+    String whiteSpaceForIndent =  "\n" + "\u0020".repeat(8) // white space for indentation
+    String textToFind = "Temporary Suppression"
+
+    noteNodes.find {
+        if (it.getTextContent().contains(textToFind)) {
+            def noteNode = it
+            Document document = suppressionDoc.getOwnerDocument()
+            Node parentNode = noteNode.getParentNode()
+            unSuppressedCves.each {
+                def cve = it
+                noteNode.appendChild(
+                        document.createTextNode(whiteSpaceForIndent + cve + " refer [Ticket]"))
+                parentNode.appendChild(
+                        document.createElement("cve")).appendChild(document.createTextNode(cve))
+            }
+            return true
+        }
+    } ?: {throw new GradleException(String.format("No notes element with text '%s'", textToFind))}()
 }
 
-static def cleanXml(Element suppressions) {
+static def cleanXml(Element suppression) {
     // Strip out all empty nodes
     XPathExpression xpathExp = XPathFactory.newInstance().newXPath().compile(
             "//text()[normalize-space(.) = '']")
 
-    NodeList emptyTextNodes = xpathExp.evaluate(suppressions, XPathConstants.NODESET) as NodeList
+    NodeList emptyTextNodes = xpathExp.evaluate(suppression, XPathConstants.NODESET) as NodeList
     if (emptyTextNodes.getLength()) {
+        println("found")
```

Comment:
Looks like debug that can now be deleted

Reply:
I'll remove this once merged